### PR TITLE
[openstack_octavia] Collect new API resources

### PR DESCRIPTION
**Pull request by cgoncalves**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [x] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: sos/report/plugins/openstack_octavia.py [hunk 1]
```diff
@@ -60,7 +66,16 @@ def setup(self):
         for loadbalancer in loadbalancers.splitlines():
             loadbalancer = loadbalancer.split()[0]
             self.add_cmd_output(
-                "openstack loadbalancer show %s" % (loadbalancer)
```
Comment (johnsom) [inline sos/report/plugins/openstack_octavia.py:63]: This is a big change. The status tree is useful for sure, but by not capturing the load balancer configuration (flavor, az, etc.) I think we are losing a lot of useful information. I'm on this fence on this one.

**Comment by cgoncalves** [comment]
> This is a big change. The status tree is useful for sure, but by not capturing the load balancer configuration (flavor, az, etc.) I think we are losing a lot of useful information. I'm on this fence on this one. 

Good point. My first thought is always that "status show" dumps everything which is not true. Do you think it is best to just do a "loadbalancer show" or both show and status?

File: sos/report/plugins/openstack_octavia.py (hunk 1 above)
Comment (cgoncalves) [reply sos/report/plugins/openstack_octavia.py:63]: Good point. My first thought is always that "status show" dumps everything which is not true. Do you think it is best to just do a "loadbalancer show" or both show and status?

Comment (cgoncalves) [reply sos/report/plugins/openstack_octavia.py:63]: I reverted this change back to "loadbalancer show". This output combined with other resource show commands should provide all needed information.

File: sos/report/plugins/openstack_octavia.py [hunk 2]
```diff
@@ -63,6 +69,15 @@ def setup(self):
...
 
+        # get details from each amphora
+        cmd = "openstack loadbalancer amphora list -f value -c id"
+        amphorae = self.exec_cmd(cmd)['output']
+        for amphora in amphorae.splitlines():
+            amphora = amphora.split()[0]
+            self.add_cmd_output(
+                "openstack loadbalancer amphora show %s" % (amphora)
+            )
+
```
Comment (TurboTurtle) [inline sos/report/plugins/openstack_octavia.py:80]: This gets repeated a number of times. Let's consolidate this and make it easier to add more resources in the future. We could do something like the `kubernetes` and `openshift` plugins here, E.G.:


```python
resources = [
    'amphora',
    'flavor',
    'flavorprofile'
   ... etc ...
]

for res in resources:
    cmd = "openstack loadbalancer %s list -f value -c id" % res
    ret = self.exec_cmd(cmd)
    if ret['status'] == 0:
        for ent in ret['output'].splitlines():
            ent = ent.split()[0]
            self.add_cmd_output(
                "openstack loadbalancer %s show %s" % (res, ent)
            )
```

For the `add_cmd_output()`, you could also pass `subdir=res` to get these outputs sorted by resource in `sos_commands/openstack_octavia/$res`

Comment (cgoncalves) [reply sos/report/plugins/openstack_octavia.py]: Agreed. I've addressed this comment. I also fixed one other issue I found with "provider capability list".

**Pull request closed without merging**